packages/server/src/fhir/repo.ts

```diff
@@ -1178,7 +1178,7 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
           // Deprecated - to be removed
           // Add compartment restriction for the access policy.
           expressions.push(new Condition('compartments', 'ARRAY_CONTAINS', policyCompartmentId, 'UUID[]'));
-        } else if (policy.criteria) {
+        } else if (policy.criteria?.startsWith(policy.resourceType + '?')) {
```

Comment:
Should we add a case for logging when an AccessPolicy has an invalid criteria so we can take stock on who's AccessPolicy's are currently broken?